### PR TITLE
ログイン認証フィルターの実装

### PR DESCRIPTION
**Pull request by tkeshun**

### CODE REVIEW

File: src/main/java/com/example/common/SecurityConfig.java [hunk 1]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
 		return http.build();
 	}
 
+	@Bean
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:33]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:33]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 2]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+	SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:34]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 3]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+		http.authorizeHttpRequests(ahr -> ahr
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:35]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:35]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 4]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.requestMatchers("/", "/register-user/to-register", "/register-user/register-user",
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:36]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 115).

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:36]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 5]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+						"/show-item/show-item-list", "/shopping-cart/show-cart")
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:37]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 104).

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:37]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 6]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.permitAll() // 「/」などのパスは全てのユーザに許可
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 7]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				// .requestMatchers("/admin/**").hasRole("ADMIN")
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 8]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				// /admin/から始まるパスはADMIN権限でログインしている場合のみアクセス可(権限設定時の「ROLE_」を除いた文字列を指定)
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:40]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 9]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				// .requestMatchers("/user/**").hasRole("USER") //
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 10]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				// /user/から始まるパスはUSER権限でログインしている場合のみアクセス可(権限設定時の「ROLE_」を除いた文字列を指定)
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:42]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 11]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.anyRequest().authenticated()); // それ以外のパスの場合はログイン認証が必要
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:43]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 12]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+
+		http.formLogin(login -> login // ログインに関する設定
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:45]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:45]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 13]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.loginPage("/") // ログイン画面に遷移させるパス(ログイン認証が必要なパスを指定してかつログインされていないとこのパスに遷移される)
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:46]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 107).

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:46]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 14]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.loginProcessingUrl("/to-login") // ログインボタンを押した際に遷移させるパス(ここに遷移させれば自動的にログインが行われる)
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:47]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 112).

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:47]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 15]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.failureUrl("/?error=true") // ログイン失敗に遷移させるパス
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:48]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 16]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.defaultSuccessUrl("/show-item/show-item-list", false) // 第1引数:デフォルトでログイン成功時に遷移させるパス
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:49]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 116).

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:49]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 17]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				// 第2引数: true :認証後常に第1引数のパスに遷移
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:50]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 18]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				// false:認証されてなくて一度ログイン画面に飛ばされてもログインしたら指定したURLに遷移
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:51]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 19]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.usernameParameter("mailAddress") // 認証時に使用するユーザ名のリクエストパラメータ名(今回はメールアドレスを使用)
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:52]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 108).

Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:52]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 20]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+				.passwordParameter("password")); // 認証時に使用するパスワードのリクエストパラメータ名
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:53]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/common/SecurityConfig.java [hunk 21]
```diff
@@ -29,6 +30,36 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
...
+
+		http.logout(logout -> logout // ログアウトに関する設定
```
Comment (tkeshun) [inline src/main/java/com/example/common/SecurityConfig.java:55]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

**Review by tkeshun — COMMENTED** [review]
Remaining comments which cannot be posted as a review comment to avoid GitHub Rate Limit

<details>
<summary>checkstyle</summary>

[src/main/java/com/example/common/SecurityConfig.java|55 col 17|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L55) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/common/SecurityConfig.java|56|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L56) Line is longer than 100 characters (found 114).
[src/main/java/com/example/common/SecurityConfig.java|56 col 1|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L56) Line contains a tab character.
[src/main/java/com/example/common/SecurityConfig.java|57|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L57) Line is longer than 100 characters (found 111).
[src/main/java/com/example/common/SecurityConfig.java|57 col 1|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L57) Line contains a tab character.
[src/main/java/com/example/common/SecurityConfig.java|58 col 1|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L58) Line contains a tab character.
[src/main/java/com/example/common/SecurityConfig.java|59|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L59) Line is longer than 100 characters (found 108).
[src/main/java/com/example/common/SecurityConfig.java|59 col 1|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L59) Line contains a tab character.
[src/main/java/com/example/common/SecurityConfig.java|60 col 1|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L60) Line contains a tab character.
[src/main/java/com/example/common/SecurityConfig.java|60 col 17|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L60) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/common/SecurityConfig.java|61 col 1|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L61) Line contains a tab character.
[src/main/java/com/example/common/SecurityConfig.java|61 col 9|](http://github.com/ecsite-project/ec-202304c/blob/d0f6e164269e15b22c1a381cd98d04e564066868/src/main/java/com/example/common/SecurityConfig.java#L61) 'method def rcurly' has incorrect indentation level 8, expected level should be 2.
</details>

**Review by yousuke-murayama — APPROVED** [review]
問題ないと思います。

**Pull request merged**